assets/dev/js/frontend/hello-frontend.js

```diff
@@ -1,33 +1,76 @@
-( function( $ ) {
-    $( document ).on( 'click', '.site-navigation-toggle', ( event ) => {
-        const $menuToggle = $( event.currentTarget ),
-              $menuToggleHolder = $menuToggle.parent( '.site-navigation-toggle-holder' ),
-              $dropdownMenu = $menuToggleHolder.siblings( '.site-navigation-dropdown' ),
-              isDropdownVisible = ! $menuToggleHolder.hasClass( 'elementor-active' );
-
-        $menuToggle.attr( 'aria-expanded', isDropdownVisible );
-        $dropdownMenu.attr( 'aria-hidden', ! isDropdownVisible );
-        $menuToggleHolder.toggleClass( 'elementor-active', isDropdownVisible );
-
-        // Always close all sub active items.
-        $dropdownMenu.find( '.elementor-active' ).removeClass( 'elementor-active' );
-    } );
-
-    $( document ).on( 'click', '.site-navigation-dropdown .menu-item-has-children > a', ( event ) => {
-        const $anchor = $( event.currentTarget ),
-              $parentLi = $anchor.parent( 'li' ),
-              isSubmenuVisible = $parentLi.hasClass( 'elementor-active' );
+class elementorHelloThemeHandler {
+    constructor() {
+        this.closeMenuItems = this.closeMenuItems.bind( this );
+
+        this.initSettings();
+        this.initElements();
+        this.bindEvents();
+    }
+
+    initSettings() {
+        this.settings = {
+            selectors: {
+                header: 'header.site-header',
+                footer: 'footer.site-footer',
+                menuToggle: '.site-header .site-navigation-toggle',
+                menuToggleHolder: '.site-header .site-navigation-toggle-holder',
+                dropdownMenu: '.site-header .site-navigation-dropdown',
+            },
+        };
+    }
+
+    initElements() {
+        this.elements = {
+            $window: jQuery( window ),
+            $document: jQuery( document ),
+            $header: jQuery( this.settings.selectors.header ),
+            $footer: jQuery( this.settings.selectors.footer ),
+            $menuToggle: jQuery( this.settings.selectors.menuToggle ),
+            $menuToggleHolder: jQuery( this.settings.selectors.menuToggleHolder ),
+            $dropdownMenu: jQuery( this.settings.selectors.dropdownMenu ),
+        };
+    }
+
+    bindEvents() {
+        this.elements.$menuToggle.on( 'click', () => {
+            const isDropdownVisible = ! this.elements.$menuToggleHolder.hasClass( 'elementor-active' );
+
+            this.elements.$menuToggle.attr( 'aria-expanded', isDropdownVisible );
+            this.elements.$dropdownMenu.attr( 'aria-hidden', ! isDropdownVisible );
+            this.elements.$menuToggleHolder.toggleClass( 'elementor-active', isDropdownVisible );
+
+            // Always close all sub active items.
+            this.elements.$dropdownMenu.find( '.elementor-active' ).removeClass( 'elementor-active' );
+
+            if ( isDropdownVisible ) {
+                this.elements.$window.on( 'resize', this.closeMenuItems );
```

Comment:
To avoid the bind statement in the `constructor`:
```suggestion
                this.elements.$window.on( 'resize', () => this.closeMenuItems() );
```

Reply:
done